### ISSUE TITLE
T-Display-AMOLED-1.75 diagram error

### ISSUE THREAD

**Issue by nikthefix** [Xinyuan-LilyGO/T-Display-S3-AMOLED-1.43#1]
The diagram on the products website shows the boot and reset buttons labelled wrongly.
The labels should be switched.

Suggestion for future similar dev boards:

Please break out the boot and reset pins to the headers so that when mounted to an expansion board or mounted in an enclosure we still get easy access to boot / reset functionality. This would be good for USB too.
Also, a rear facing USB C receptacle (like on the T-Encoder-Pro) would e more practical for panel mounting.

Thanks
nik

**Comment by Llgok** [comment]
@nikthefix 
Hello, I have checked and indeed, the BOOT button and the reset button are labeled incorrectly on the official website's store. I have already informed the relevant person in charge to make the necessary changes. Thank you for pointing this out. It is a great suggestion to bring out the BOOT pin and the reset pin to a socket, but the expansion socket is already full and cannot accommodate additional pins. If you are making a casing, you can use soldered jumper wires to install the BOOT button and the reset button onto the back panel of the casing.

**Comment by d3mac123** [comment]
@nikthefix can you do your magic (like you did for the S3 Long) and make this one work with TFT_ESPI? 🙏

**Comment by nikthefix** [comment]
@d3mac123 

Hello again. Actually I'm just working on exactly that. The 1.75" and 1.43" Lilygo round amoleds use different driver IC's (different from each other) and different instruction sets (mainly just in the init sequences). I'm preparing bare bones drivers for both to put on my github but in the meantime I can tell you that the process for TFT_eSPI compatibility is exactly the same as for the S3 Long - use TFT_eSPI to populate a full screen sprite then push all pixels from the sprite buffer to the display driver frame buffer. The Lilygo library gives you what you need to do this.

These new QSPI amoled displays we're seeing more and more of are, I guess, MIPI at heart but have had 'maker friendly QSPI' thrust upon them. I'm glad. But it means that the sometimes the coding will differ from TFT. For example, there's no backlight to energize but instead a 'display enable'.

I'll get back to you with updates.

nik

**Comment by d3mac123** [comment]
I'm looking forward to see the 1.75 specifically. I'm building an adsb in receiver (code is working already iin the S3) for the plane and the round shape will be amazing.

Alex

**Comment by nikthefix** [comment]
@d3mac123 Well I have some 1.75" code I can send you right now but it's LVGL. My driver is separate tho so you could use that.

**Comment by d3mac123** [comment]
No need for lgvl code for now. I keep trying to avoid learning them 🙃

Alex

**Comment by nikthefix** [comment]
@d3mac123 OK I'll post a hello world for it using TFT_eSPI

**Comment by nikthefix** [comment]
@d3mac123 

Here's an LVGL / SLS example just for testing purposes:

https://github.com/nikthefix/Lilygo_Support_T_Display_Amoled_1.75inch_round

It shouldn't require any library setup as all the libs are local - including LVGL  / lv_conf.h

TFT_eSPI example to follow. 

nik

**Comment by d3mac123** [comment]
@nikthefix 

I am still waiting for the board to be delivered. Hopefully, you will have the TFT_eSPI example when it gets here. My code is ready to be tested. I appreciate all your work. Thank you so much for that.

**Comment by nikthefix** [comment]
@d3mac123 

Oh is see! Haha!

It's a great board - you'll love it.
1.75" is the perfect size for circular and a nicer alternative to the 2.1" ST7701 ips IMO.

nik

**Comment by d3mac123** [comment]
@nikthefix Board just arrived. Looking forward to your magical code :)

**Comment by kirchgatterj** [comment]
@nikthefix I tried applying your example to a screen I built, but the colors seem to be off. Any ideas?

**Comment by kiddingme1** [comment]
Looking forward to your code as well @nikthefix! Loved the work you did to get the rm67162 working with TFT_espi. Currently have some demo code running on the new 1.75" but has bad tearing.

**Comment by d3mac123** [comment]
It is probably some stupidity from my side but I am getting the following error:
```
In file included from /Users/alex/Documents/Arduino/1.75inch_amoled_nik_local_libs/1.75inch_amoled_nik_local_libs.ino:67:
/Users/alex/Documents/Arduino/1.75inch_amoled_nik_local_libs/src/ui/ui.h:13:10: fatal error: ..\lvgl\lvgl.h: No such file or directory
   13 | #include "..\lvgl\lvgl.h"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
exit status 1

Compilation error: ..\lvgl\lvgl.h: No such file or directory

```
Any ideas of what I may be doing wrong?

**Comment by nikthefix** [comment]
@kirchgatterj 
Is that the lilygo 1.75" board or the lilygo 1.43" board or a custom board?
Is the demo using LVGL or TFT_eSPI?
If LVGL then it looks like in lv_conf.h you need to #define LV_COLOR_16_SWAP 1
If TFT_eSPI then look for colour swap in the user_setup file.


The tearing issue is solvable as thankfully Lilyo did break out the TE pin (which they didn't do on the T4-S3).
The tearing should only be apparent when using non-native display orientation.

There is a whole thread on anti-tearing on the T-Display-S3 which I will link here when I find it. I think it will be relevant.

nik

**Comment by kirchgatterj** [comment]
It is the 1.75 and I am using LVGL. I will take a look at this link and see if I can make any sense out of it. If you are planning on integrating that into your driver that driver that would be amazing!

**Comment by kirchgatterj** [comment]
Setting  LV_COLOR_16_SWAP 1 does not resolve my issue.

**Comment by nikthefix** [comment]
@kirchgatterj Does my demo work out-of-the-box?

**Comment by kirchgatterj** [comment]
No, I have the same issue as the compiler claims the pin_config.h files is missing, etc.

**Comment by nikthefix** [comment]
OK, I'll post another version using the standard Arduino folder structure layout. It seems that my experimental layout has caused some problems.

**Comment by kirchgatterj** [comment]
It looks to be an issue with transparencies...

The line on the right is supposed to be the yellow needle that is wider on the top and thinner on the bottom and has transparencies.

![Image](https://github.com/user-attachments/assets/d65efad8-5555-478c-8413-3f20f09df570)

**Comment by nikthefix** [comment]
@kirchgatterj I uploaded a simpler version of my demo using LVGL / lv_conf.h in the usual Libraries directory:

Please try to see if the colours look correct:

https://github.com/nikthefix/Lilygo_Support_T_Display_Amoled_1.75inch_round

nik

**Comment by slash-bit** [comment]
Thanks @nikthefix  to get me started with T-Display S3 1.75 AMOLED round. I was loosing hope with Arduino_GFX, the quality of Text circles was really poor. Now with TFT_eSPI its a different story.
I am re-building an aviation proximity display (Radar) based on this round display and looks great.
https://github.com/slash-bit/SkyView-AMOLED/blob/main/README.md

![Image](https://github.com/user-attachments/assets/e48291ff-b9e4-496d-a991-7639b328cec7)

![Image](https://github.com/user-attachments/assets/01c58fc8-a346-4ecc-8ffa-804a9c7b5b39)
